valohai_yaml/objs/task.py

```diff
@@ -0,0 +1,109 @@
+from enum import Enum
+from typing import Any, List, Optional, Union
+
+from valohai_yaml.objs.base import Item
+from valohai_yaml.objs.utils import (
+    check_type_and_listify,
+    consume_array_of,
+)
+
+
+class TaskParameterType(Enum):
+    """Represents a task parameter style definition."""
+
+    BAYESIAN_TPE = 'bayesian_tpe'
+    GRID_SEARCH = 'grid_search'
+    DISTRIBUTED = 'distributed'
+    RANDOM_SEARCH = 'random_search'
+    MANUAL_SEARCH = 'manual_search'
+
+    @classmethod
+    def cast(cls, value: Optional[Union['TaskParameterType', str]]) -> 'TaskParameterType':
+        if not value:
+            return TaskParameterType.GRID_SEARCH
+        if isinstance(value, TaskParameterType):
+            return value
+        value = str(value).lower()
+        return TaskParameterType(value)
+
+
+class VariantParameterStyle(Enum):
+    """Represents a variant parameter style definition."""
+
+    LOGSPACE = 'logspace'
+    MULTIPLE = 'multiple'
+    LINEAR = 'linear'
+    SINGLE = 'single'
+    RANDOM = 'random'
+
+    @classmethod
+    def cast(cls, value: Optional[Union['VariantParameterStyle', str]]) -> 'VariantParameterStyle':
+        if not value:
+            return VariantParameterStyle.SINGLE
+        if isinstance(value, VariantParameterStyle):
+            return value
+        value = str(value).lower()
+        return VariantParameterStyle(value)
+
+
+class VariantParameter(Item):
+    """Represents a variant parameter definition."""
+
+    rules: Any
```

Comment:
I think we'll know `rules` will always be a dict with string keys, so
```suggestion
    rules: dict[str, Any]
```